### DISCUSSION TITLE
Getting this error : [nodemon] app crashed - waiting for file changes before starting...

### DISCUSSION THREAD

**Discussion by sabeehsharif** [microsoft/hack-together-teams#41] [Q&A]
Hi @waldekmastykarz / @aycabas 

Getting this error on running project in Terminal:
[nodemon] app crashed - waiting for file changes before starting...

i have restarted Node server and vs code but still error exists. 
![image](https://github.com/microsoft/hack-together-teams/assets/50944211/f5b9940e-289a-4aa7-85f7-4f2e6e069f4a)

**Comment by waldekmastykarz** [comment]
Thanks for sharing it with us @sabeehsharif. Do you have any more information for us so that we can try and reproduce it?

**Reply by aycabas** [reply]
Also @sabeehsharif when I get the app crashed error I usually get other error messages under the **Problems** section as well, either an issue in one of the project files or unsaved file in the project root. Can you please check the **Problems** section and let us know if there is any error there?

**Reply by sabeehsharif** [reply]
Hi @waldekmastykarz and @aycabas 
Basically issue is with OpenAI Api key, when i set hard coded it works fine but once saved as Environmental variable then i get this error. 
creating variable like this : $env:OPENAI_API_KEY='sfdsfsdfsdferwerwe'

**Reply by waldekmastykarz** [reply]
@sabeehsharif when you create it like this, isn't it bound to the current PowerShell session only and when you start your app, it's running in a different session and the env var is no longer available? Have you tried setting it up using your machine's settings?

**Reply by sabeehsharif** [reply]
Hi @waldekmastykarz 

yes i tried by creating global environmental variable in advance setting via control panel. Didn’t work

**Reply by waldekmastykarz** [reply]
If you go to the terminal and execute `node -e "console.log(process.env)"` do you see your environment variable listed there?

**Reply by sabeehsharif** [reply]
Hi @waldekmastykarz ,

yes i can see the variables
![image](https://github.com/microsoft/hack-together-teams/assets/50944211/0c9b0c29-411f-42bb-9099-9d85cf8178cf)

**Reply by sabeehsharif** [reply]
This is complete detail of error
![image](https://github.com/microsoft/hack-together-teams/assets/50944211/6565dfe9-cdee-4d72-b2c5-1783eacad628)

**Reply by waldekmastykarz** [reply]
Do you have your code somewhere where we could look at it?

**Reply by sabeehsharif** [reply]
@waldekmastykarz  and @aycabas 

Thank you so much for your support it's fixed. Submitted Hack.

**Reply by waldekmastykarz** [reply]
What was the issue in the end in case someone else comes across it too?
